pallets/parachain-staking/src/weights.rs

```diff
@@ -124,275 +121,285 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+		Weight::from_ref_time(28_958_000 as u64)
+			.saturating_add(T::DbWeight::get().reads(1 as u64))
+			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: ParachainStaking CollatorCommission (r:1 w:1)
 	#[rustfmt::skip]
 	fn set_collator_commission() -> Weight {
-		(17_036_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+		Weight::from_ref_time(24_841_000 as u64)
+			.saturating_add(T::DbWeight::get().reads(1 as u64))
+			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: ParachainStaking TotalSelected (r:1 w:0)
 	// Storage: ParachainStaking InflationConfig (r:1 w:1)
 	#[rustfmt::skip]
 	fn set_blocks_per_round() -> Weight {
-		(57_722_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
-			.saturating_add(T::DbWeight::get().writes(1 as Weight))
+		Weight::from_ref_time(65_362_000 as u64)
+			.saturating_add(T::DbWeight::get().reads(2 as u64))
+			.saturating_add(T::DbWeight::get().writes(1 as u64))
 	}
 	// Storage: ParachainStaking CandidateInfo (r:1 w:1)
 	// Storage: ParachainStaking DelegatorState (r:1 w:0)
 	// Storage: ParachainStaking CandidatePool (r:1 w:1)
 	// Storage: System Account (r:1 w:1)
+	// Storage: Balances Locks (r:1 w:1)
 	// Storage: ParachainStaking Total (r:1 w:1)
 	// Storage: ParachainStaking TopDelegations (r:0 w:1)
 	// Storage: ParachainStaking BottomDelegations (r:0 w:1)
 	#[rustfmt::skip]
 	fn join_candidates(x: u32, ) -> Weight {
-		(70_223_000 as Weight)
+		Weight::from_ref_time(90_562_000 as u64)
 			// Standard Error: 1_000
-			.saturating_add((69_000 as Weight).saturating_mul(x as Weight))
-			.saturating_add(T::DbWeight::get().reads(5 as Weight))
-			.saturating_add(T::DbWeight::get().writes(6 as Weight))
+			.saturating_add(Weight::from_ref_time(155_000 as u64).saturating_mul(x as u64))
+			.saturating_add(T::DbWeight::get().reads(6 as u64))
+			.saturating_add(T::DbWeight::get().writes(7 as u64))
 	}
 	// Storage: ParachainStaking CandidateInfo (r:1 w:1)
 	// Storage: ParachainStaking CandidatePool (r:1 w:1)
 	#[rustfmt::skip]
 	fn schedule_leave_candidates(x: u32, ) -> Weight {
-		(53_351_000 as Weight)
+		Weight::from_ref_time(73_297_000 as u64)
 			// Standard Error: 1_000
-			.saturating_add((66_000 as Weight).saturating_mul(x as Weight))
-			.saturating_add(T::DbWeight::get().reads(2 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			.saturating_add(Weight::from_ref_time(132_000 as u64).saturating_mul(x as u64))
+			.saturating_add(T::DbWeight::get().reads(2 as u64))
+			.saturating_add(T::DbWeight::get().writes(2 as u64))
 	}
 	// Storage: ParachainStaking CandidateInfo (r:1 w:1)
 	// Storage: ParachainStaking TopDelegations (r:1 w:1)
-	// Storage: System Account (r:2 w:2)
 	// Storage: ParachainStaking DelegatorState (r:1 w:1)
+	// Storage: Balances Locks (r:2 w:2)
```

Comment:
This seems to be quite under-estimated now, given that your new results measure 1 more db read and 1 more db write per candidate